### PR TITLE
System Tests

### PR DESCRIPTION
**Pull request by lipmas**
## Description of the change

Adds end-to-end system tests utilizing the zli.
 
## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-1243

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: src/cli-driver.ts [hunk 1]
```diff
@@ -581,7 +592,11 @@ export class CliDriver
                     await oauth.getIdTokenAndExitOnError();
                 }
             )
-            .option('configName', {type: 'string', choices: ['prod', 'stage', 'dev'], default: this.envMap['configName'], hidden: true})
+            .option('configName', {type: 'string', choices: ['prod', 'stage', 'dev', 'system-test'], default: envMap.configName, hidden: true})
```
Comment (lipmas) [inline src/cli-driver.ts:595]: `system-test` config is no longer used and should be removed

File: src/cli-driver.ts [hunk 2]
```diff
@@ -609,7 +624,8 @@ Need help? https://cloud.bastionzero.com/support`)
                         } else {
                             this.logger.error(err.message);
                             if (err.stack)
-                                this.logger.debug(err.stack);
+                                console.log(err.stack);
+                            this.logger.debug(err.stack);
```
Comment (lipmas) [inline src/cli-driver.ts:628]: revert this

File: package.json [hunk 3]
```diff
@@ -62,12 +65,13 @@
...
     "@types/tcp-port-used": "^1.0.0",
     "@types/term-size": "^2.0.3",
-    "@types/yargs": "^15.0.10",
+    "@types/yargs": "^17.0.5",
     "async": "^3.2.0",
     "chalk": "^4.1.2",
     "cli-table3": "^0.6.0",
     "conf": "^7.1.2",
     "crypto-js": "^4.0.0",
+    "digitalocean-js": "^1.5.9",
```
Comment (ymarcus93) [inline package.json:74]: Move to `devDependencies`

File: src/system-tests/digital-ocean/digital-ocean-ssm-target-service.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            name: parameters.dropletName,
+            region: parameters.dropletRegion,
+            size: parameters.dropletSize,
+            image: parameters.dropletImage,
+            user_data: parameters.userDataScript,
+            tags: parameters.dropletTags,
+            // Key ID for system-test SSH key that exists on our account. This
+            // parameter is required when using custom images (e.g. AL2). Find
+            // the key ID of SSH keys using: doctl compute ssh-key list
+            ssh_keys: ['32360322']
```
Comment (ymarcus93) [inline src/system-tests/digital-ocean/digital-ocean-ssm-target-service.ts:176]: use fingerprint instead

**Review by lipmas — COMMENTED** [review]
Reviewed LGTM

**Review by ymarcus93 — APPROVED** [review]
LGTM

**Pull request merged**